### PR TITLE
Rename forecasts to time series

### PR DESCRIPTION
**Pull request by daniel-thom**
This PR primarily renames `forecast` to `time_series`.  It also does the following:
- Renames `iterate_forecasts` to `get_time_series_multiple`.
- Extends the former `get_forecast_values` into `get_time_series_arrays`, `get_time_series_timestamps`, and `get_time_series_values`.  This allows users to get the underlying TimeArray, timestamps, or values.
- Renames `scaling_factor` to `normalization_factor`.
- Adds `scaling_factor_multiplier` to all time series.

This requires the PowerSystems branch of the same name for cross-project testing to pass.

### CODE REVIEW

File: src/generated/PiecewiseFunction.jl [hunk 1]
```diff
@@ -2,31 +2,35 @@
 This file is auto-generated. Do not edit.
 =#
 """
-    mutable struct PiecewiseFunction <: Forecast
+    mutable struct PiecewiseFunction <: TimeSeriesData
```
Comment (jd-lara) [inline src/generated/PiecewiseFunction.jl:5]: When did we implement this struct? It seems redundant to what we are planning on doing for the Variable Cost.

File: src/generated/ScenarioBased.jl [hunk 2]
```diff
@@ -2,31 +2,35 @@
 This file is auto-generated. Do not edit.
 =#
 """
-    mutable struct ScenarioBased <: Forecast
+    mutable struct ScenarioBased <: TimeSeriesData
```
Comment (jd-lara) [inline src/generated/ScenarioBased.jl:5]: We can rename ScenatioBased to just Scenarios. @claytonpbarrows what do you think? I find awkward `ScenarioBased` time series.

File: src/system_data.jl [hunk 3]
```diff
@@ -51,294 +51,319 @@ function SystemData(;
...
     )
 end
 
 """
-Adds forecasts from a metadata file or metadata descriptors.
+Adds time_series from a metadata file or metadata descriptors.
 
 # Arguments
-- `::Type{T}`: forecasted component type; may be abstract
+- `::Type{T}`: time_seriesed component type; may be abstract
```
Comment (jd-lara) [inline src/system_data.jl:81]: typo?

**Comment by jd-lara** [comment]
The PR looks great in terms of new capabilities and implementation. We will probably need to test drive it more. I have a comment - question about the PWL forecast object. We should define if we want to keep it in 1.0 or not.

File: src/generated/ScenarioBased.jl (hunk 2 above)
Comment (claytonpbarrows) [reply src/generated/ScenarioBased.jl:5]: I also find it to be awkward. But I don't love `Scenarios` either. Also, with `TimeSeriesData` do we even need `ScenarioBased` now that we have both `label` and `accessor`? If the answer is yes, then I'm fine with the proposal.

File: src/generated/PiecewiseFunction.jl (hunk 1 above)
Comment (daniel-thom) [reply src/generated/PiecewiseFunction.jl:5]: https://github.com/NREL-SIIP/InfrastructureSystems.jl/pull/90

Comment (daniel-thom) [reply src/generated/PiecewiseFunction.jl]: Removed.

**Review by jd-lara — APPROVED** [review]
LGTM

**Pull request merged**